### PR TITLE
ASC updates

### LINKED ISSUE #1889: Shortened links not repo friendly
When I check prs, I can't check the shortened links easily. I have to add `https://osu.ppy.sh` prefix and remove the github url. Probably an ASC change.

### LINKED ISSUE #1996: Image file name spacing should have a preferred character
I prefer the hyphen(`-`), since article titles already have the underscore(`_`).

### PR DESCRIPTION
**Pull request by MegaApplePi**
Update some more stuff that has changed overtime, add headings to some stuff I didn't know was called, and other changes.

- ~~Add "soft breaks" heading (the single sentence line thing)~~
- ~~Add "line breaks" heading (the double whitespace thing)~~
- ~~Include image examples for the no-, line-, and soft breaks~~
- Escape only when needed
- Try to keep headings plural (thematic break -> thematic breaks)
- Add hatnote in trademarks section to see the osu! section (since osu! is also a trademark, but has its own section)
- Add "created" to the terminology list
- Note that osu! can be used as the game mode name for certain articles, like the RC
- Various wording changes
- Other internal links must include the hostname (`https://osu.ppy.sh/` part of the URL) -- resolves #1889 
- Replace spaces with hyphens in file names (previously: use either or underscores/hyphens) -- resolves #1996 
- Rewrite the line endings into paragraphs
- More various wording changes

---

### CODE REVIEW

File: wiki/Article_Styling_Criteria/en.md [hunk 1]
```diff
@@ -192,24 +192,55 @@ Skinning allows anyone to change the way osu! looks and feels.
...
 
-#### Whitespace
+```md
+Skinning allows anyone to change the way osu! looks and feels. This can be as small as changing the cursor or as big as redoing the appearances of all the game modes and interface.
+```
 
-Appending two trailing spaces at the end of a line is parsed as a line break. (Where this is done, the paragraph will break into a separate line but be part of the same paragraph.) Line breaks are not allowed in the osu! wiki.
+And both will look like this:
+
+![](img/soft-breaks.png "Although the text is in two sepatate lines, it is parsed as a single paragraph.")
```
Comment (Joehuu) [inline wiki/Article_Styling_Criteria/en.md]:
```suggestion
![](img/soft-breaks.png "Although the text is in two separate lines, it is parsed as a single paragraph.")
```

File: wiki/Article_Styling_Criteria/en.md [hunk 2]
```diff
@@ -349,9 +382,15 @@ The markup for code is a grave mark (`` ` ``). To put a grave mark in code, surr
 
 #### Keyboard keys
 
-When doing this, use capital letters for single characters and title case for modifiers. Denoting the letter itself, not the keyboard key, should not be in code (it should use quotes).
+When doing this, use capital letters for single characters and title case for modifiers. Denoting the letter itself, not the keyboard key, should not be in code (it should use quotes). For example:
+
+```md
+pippi is spelt with a lower case "P" like peppy.
```
Comment (Joehuu) [inline wiki/Article_Styling_Criteria/en.md]:
```suggestion
pippi is spelt with a lowercase "P" like peppy.
```

File: wiki/Article_Styling_Criteria/en.md [hunk 3]
```diff
@@ -304,19 +335,21 @@ Optionally, add an extra line before and after the heading to help make the head
...
 
 #### Sections
 
-All section headings must use levels 2 to 5, never level 6. Section headings must use sentence case and should never contain styles nor links. In addition to this, section headings must never skip a heading level (e.g. do not go from a level 2 heading to a level 4 heading). The first section heading must be a level 2 heading. Unlike titles, section headings can have small image icons.
+All section headings must use levels 2 to 5, never level 6. Section headings must use sentence case and must not contain formatting nor links. In addition to this, section headings must not skip a heading level (e.g. do not go from a level 2 heading to a level 4 heading). The section heading after the title heading must be a level 2 heading. Unlike titles, section headings can have small image icons.
 
-In the osu! wiki, heading levels 4 and above do not appear in the table of contents and are not given the id attribute (this means you cannot link directly to these headings).
+In the osu! wiki, heading levels 4 to 6 do not appear in the table of contents and are not given the id attribute (these cannot be linked directly).
```
Comment (Joehuu) [inline wiki/Article_Styling_Criteria/en.md]: > All section headings must use levels 2 to 5.
```suggestion
In the osu! wiki, heading levels 4 and 5 do not appear in the table of contents and are not given the id attribute (these cannot be linked directly).
```

File: wiki/Article_Styling_Criteria/en.md [hunk 4]
```diff
@@ -304,19 +335,21 @@ Optionally, add an extra line before and after the heading to help make the head
...
+---
 
-For English article titles, if you need to reword the title, you must rename the folder to match the article title. There are rare exceptions to this however.
+For English article titles, if the article's title needs to be renamed, the folder name it is in must also be renamed to match it.
 
 #### Sections
 
-All section headings must use levels 2 to 5, never level 6. Section headings must use sentence case and should never contain styles nor links. In addition to this, section headings must never skip a heading level (e.g. do not go from a level 2 heading to a level 4 heading). The first section heading must be a level 2 heading. Unlike titles, section headings can have small image icons.
+All section headings must use levels 2 to 5, never level 6. Section headings must use sentence case and must not contain formatting nor links. In addition to this, section headings must not skip a heading level (e.g. do not go from a level 2 heading to a level 4 heading). The section heading after the title heading must be a level 2 heading. Unlike titles, section headings can have small image icons.
```
Comment (Joehuu) [inline wiki/Article_Styling_Criteria/en.md]: Do not need to mention as there is a range stated.
```suggestion
All section headings must use levels 2 to 5. Section headings must use sentence case and must not contain formatting nor links. In addition to this, section headings must not skip a heading level (e.g. do not go from a level 2 heading to a level 4 heading). The section heading after the title heading must be a level 2 heading. Unlike titles, section headings can have small image icons.
```

File: wiki/Article_Styling_Criteria/en.md [hunk 5]
```diff
@@ -429,33 +468,26 @@ Do not: [https://osu.ppy.sh/home](https://osu.ppy.sh/home)
 Do:     [osu! on the web](https://osu.ppy.sh/home)
 ```
 
-#### Internal
+#### Internal links
 
-Internal links are the links linking within the website.
+*Note: Internal links refers to links that stay inside the `https://osu.ppy.sh/` domain.*
```
Comment (Joehuu) [inline wiki/Article_Styling_Criteria/en.md]:
```suggestion
*Note: Internal links refer to links that stay inside the `https://osu.ppy.sh/` domain.*
```

File: wiki/Article_Styling_Criteria/en.md [hunk 6]
```diff
@@ -918,17 +949,17 @@ Use `e.g.` when initialising "for example" and use `i.e.` for "that is".
...
+Avoid: BNs can nominate your beatmapset.
 
-Do:     Members of the Beatmap Nominators can nominate your beatmapset.
+Do:    Members of the Beatmap Nominators can nominate your beatmapset.
 ```
 
 ### Date formatting
 
-Since many regions use different date and time formats, the osu! wiki will only use the [ISO-8601](https://en.wikipedia.org/wiki/ISO_8601) format, which is `YYYY-MM-DD`. For example:
+The osu! wiki will use the [ISO-8601](https://en.wikipedia.org/wiki/ISO_8601) format, which is `YYYY-MM-DD`. For example:
```
Comment (Joehuu) [inline wiki/Article_Styling_Criteria/en.md]:
```suggestion
The osu! wiki uses the [ISO-8601](https://en.wikipedia.org/wiki/ISO_8601) format, which is `YYYY-MM-DD`. For example:
```

File: wiki/Article_Styling_Criteria/en.md (hunk 3 above)
Comment (MegaApplePi) [reply wiki/Article_Styling_Criteria/en.md]: I thought it should still be mentioned even if it is not to be used, but I suppose it might lead to some confusion.

**Comment by Joehuu** [comment]
"Soft breaks" seem to behave differently on web.

[Source](https://github.com/ppy/osu-wiki/blob/master/wiki/BBCode/en.md#heading-v1)

GitHub:



Code (no two spaces):



Web:

**Comment by Joehuu** [comment]
Anyway, I don't think the ASC should have things we shouldn't do *or* showing behavior of markdown.

It should just have the correct way imo. Doing this will shorten the article drastically.

**Comment by MegaApplePi** [comment]
It looks like that was changed in ppy/osu-web#4121 to use a line break instead.

**Comment by cl8n** [comment]
I feel like we should avoid those anyway, there are very few opportunities where it can be used well (usually you just want a new paragraph)

**Comment by Joehuu** [comment]
Let's save that for another time.

What I mean is content with `never` in it. We do not need to mention other wrong format or explain the behavior. Also there is some assuming which will not be always the case.

For example, we do not need to explain breaks and say never use them. We can just say "use one empty line per paragraph and use a one empty line padding for more paragraphs".

Assuming examples:
> ~Never delete translated articles~
> ~never set the boolean to `false`~

TL;DR: The ASC should only include the right ways to format the wiki as it is:
> enforced styling standards

and not a guide on markdown :D.

**Comment by cl8n** [comment]
oh I see, sorry I misread the conversation I thought it was actually about that particular rule

yeah, I agree with Joehuu then

**Comment by Joehuu** [comment]
This resolves #1889.

File: wiki/Article_Styling_Criteria/en.md [hunk 7]
```diff
@@ -556,7 +586,7 @@ Where `<input>` is the file name to be compressed and `<output>` is the compress
 
 *Notice: All image file extensions must use lowercase letters, otherwise they will fail to load!*
 
-Image file names must never use spaces (use underscores (`_`) or hyphens (`-`) instead). When naming an image, the file name should be meaningful or descriptive but short. Never use timestamps or long hashes for the file name.
+Image file names must not use spaces (use underscores (`_`) or hyphens (`-`) instead). When naming an image, the file name should be meaningful or descriptive but short. Never use timestamps or long hashes for the file name.
```
Comment (Joehuu) [inline wiki/Article_Styling_Criteria/en.md]: Can you prefer a character for the spacing to resolve #1996.

Comment (MegaApplePi) [reply wiki/Article_Styling_Criteria/en.md]: Alright, we'll go with hyphens, for [this reason](https://github.com/ppy/osu-wiki/issues/1996#issuecomment-457861162)?

Comment (Joehuu) [reply wiki/Article_Styling_Criteria/en.md]: Also the OP's reason, yeah.

File: wiki/Article_Styling_Criteria/en.md [hunk 8]
```diff
@@ -570,40 +487,32 @@ Installing osu! is easy. First, download the installer from the download page.
...
-The osu! wiki will give every image a max-width, being the width of the article body (680 pixels). This means when an image's width is more than 680 pixels, you should consider downscaling it to 680 pixels to limit wasted digital space.
+The osu! wiki gives every image a max-width, being the width of the article body (680 pixels). When an image's width is more than 680 pixels, consider downscaling it.
 
 #### Annotating images
 
 When annotating images, use *Exo 2 light*. For Chinese, Korean, Japanese characters, use *Microsoft YaHei*.
 
 Try to limit writing text in images, as it is difficult for translators (and other editors) to edit them.
 
-##### Translating annotated images
```
Comment (Joehuu) [inline wiki/Article_Styling_Criteria/en.md]: Any reason removing this? It explains a proper format.

This can surely be used when taking pictures of the osu! client, which has locales/translations.

File: wiki/Article_Styling_Criteria/en.md [hunk 9]
```diff
@@ -655,15 +562,15 @@ All of the flag icons use the two letter code (in all capital letters) and end w
 ![](/wiki/shared/flag/xx.gif)
 ```
 
-`xx` is the [ISO 3166-2](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2 "Wikipedia") two-lettered country code for the flag.
+Where `xx` is the [ISO 3166-2](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2 "Wikipedia") two-lettered country code for the flag.
 
-Adding the full country names in the alt text is optional.
+Adding the full country names or codes in the alt text is optional. Adding full country names in title text is optional.
```
Comment (Joehuu) [inline wiki/Article_Styling_Criteria/en.md]:
```suggestion
Adding the full country names or codes in the alternate text is optional. Adding full country names in title text is optional.
```

File: wiki/Article_Styling_Criteria/en.md [hunk 10]
```diff
@@ -727,7 +628,7 @@ For example, use:
 
 ### osu!
 
-The name of the game, osu!, is never capitalised nor italicised. The osu! official branding must not use any spaces. Examples include:
+The name of the game, osu!, must not capitalised nor italicised. The osu! official branding must not use any spaces. Examples include:
```
Comment (Joehuu) [inline wiki/Article_Styling_Criteria/en.md]:
```suggestion
The name of the game, osu!, must not be capitalised nor italicised. The osu! official branding must not use any spaces. Examples include:
```

File: wiki/Article_Styling_Criteria/en.md [hunk 11]
```diff
@@ -834,31 +735,33 @@ The `#spanish` chat channel are for those who speak Spanish.
...
+Proper nouns must be first-letter capitalised. For example:
 
 ```
 Dean Herbert (also known as peppy) created osu!.
 ```
 
 #### Trademarks
 
-The following trademarks must be spelt as follows (note the letter casing):
+*For rules regarding osu!, see [osu!](#osu!).*
```
Comment (Joehuu) [inline wiki/Article_Styling_Criteria/en.md]:
```suggestion
*For rules regarding osu!, see: [osu!](#osu!)*
```

File: wiki/Article_Styling_Criteria/en.md [hunk 12]
```diff
@@ -867,34 +770,38 @@ When writing the name of the game modes, they must be written as follows:
...
 - `game mode` (or `mode`)
 - `game mod` (or `mod`)
 - `gameplay`
 - `hit burst`
-- `hit circles`
-- `hit sounds`
+- `hit circle`
+- `hit sound`
 - `in-game`
 - `play style`
```
Comment (Joehuu) [inline wiki/Article_Styling_Criteria/en.md]: Either use the suggestion below (idk why it does this on non-diffs) or commit another way.
```suggestion
- `playstyle`
```

File: wiki/Article_Styling_Criteria/en.md [hunk 13]
```diff
@@ -867,34 +770,38 @@ When writing the name of the game modes, they must be written as follows:
...
 
 ---
 
-- Use `beatmaps` instead of `maps`.
+- Use `beatmap` instead of `map`.
 - Use `creator` instead of `beatmapper` or `mapper`.
-- Use `BN` or `Beatmap Nominators` when referring to the *Beatmap Nominators*. Never use `BNG` or `Beatmap Nomination Group`.
-- Use `sign in` instead of `log in`.
-- Use `sign out` instead of `log out`.
+- Use `created` instead of `beatmapped` or `mapped`.
```
Comment (Joehuu) [inline wiki/Article_Styling_Criteria/en.md]: Still curious as to why we don't follow `osu-web`'s terminology with `mapped`.

File: wiki/Article_Styling_Criteria/en.md [hunk 14]
```diff
@@ -146,74 +144,35 @@ Parts of the GFM (GitHub Flavored Markdown) markup are supported in the osu! wik
 
 *For more information about HTML entities, see: [Entity - Glossary](https://developer.mozilla.org/en-US/docs/Glossary/Entity "MDN")*
 
-HTML character entities should almost never be used. If used however, prefer the character over the entity. For example:
-
-```
-Do not: all this with more transparency&trade; than ever before!
-Do not: all this with more transparency&#153; than ever before!
-Do not: all this with more transparency&#x99; than ever before!
+HTML character entities should not be used. If used however, prefer the character over the entity. For example:
```
[the remark this answers is not in the document]
Comment (MegaApplePi) [reply wiki/Article_Styling_Criteria/en.md]: > I don't think the [...]

Huh, didn't think that they were part of HTML despite having it in the name. Fwiw, I copy it from Google or use amp-what.

> We can add special case sections [...]

Oh yeah, any thoughts on the wording for this section? I'm going to go along the lines of: `HTML comments may be used to annotate text. Keep usage limited.`

File: wiki/Article_Styling_Criteria/en.md (hunk 13 above)
Comment (MegaApplePi) [reply wiki/Article_Styling_Criteria/en.md]: It was to go with the term above "creator", will change it to "mapped" instead.

File: wiki/Article_Styling_Criteria/en.md (hunk 12 above)
Comment (MegaApplePi) [reply wiki/Article_Styling_Criteria/en.md]: We are going with "playstyle" now?

Comment (MegaApplePi) [reply wiki/Article_Styling_Criteria/en.md]: Never mind, looked back at the Discord again.

**Comment by Joehuu** [comment]
Can you fix `Metadata` section's heading level? It should be level 3.

File: wiki/Article_Styling_Criteria/en.md [hunk 15]
```diff
@@ -598,10 +509,10 @@ Try to limit writing text in images, as it is difficult for translators (and oth
 
 ##### Translating annotated images
 
-When translating annotated images, the localised image version must be placed in the same directory as the original version (i.e. the English version). The filename of a localised image version must start with the original version's name, followed by a hyphen, followed by the locale name (in caps). For example:
+When translating annotated images, the localised image version must be placed in the same directory as the original version (i.e. the English version). The filename of a localised image version must start with the original version's name, followed by a hyphen, followed by the locale name (in caps). For example:
```
Comment (Joehuu) [inline wiki/Article_Styling_Criteria/en.md]:
```suggestion
When translating annotated images, the localised image version must be placed in the same directory as the original version (i.e. the English version). The filename of a localised image version must start with the original version's name, followed by a hyphen, followed by the locale name (in caps). For example:
```

File: wiki/Article_Styling_Criteria/en.md [hunk 16]
```diff
@@ -598,10 +509,10 @@ Try to limit writing text in images, as it is difficult for translators (and oth
 
 ##### Translating annotated images
 
-When translating annotated images, the localised image version must be placed in the same directory as the original version (i.e. the English version). The filename of a localised image version must start with the original version's name, followed by a hyphen, followed by the locale name (in caps). For example:
+When translating annotated images, the localised image version must be placed in the same directory as the original version (i.e. the English version). The filename of a localised image version must start with the original version's name, followed by a hyphen, followed by the locale name (in caps). For example:	
 
-- `hardrock_mod_vs_easy_mod.jpg` (i.e. in English)
-- `hardrock_mod_vs_easy_mod-DE.jpg` (i.e. in German)
+- `hardrock_mod_vs_easy_mod.jpg` (i.e. in English)
```
Comment (Joehuu) [inline wiki/Article_Styling_Criteria/en.md]:
```suggestion
- `hardrock_mod_vs_easy_mod.jpg` (i.e. in English)
```

File: wiki/Article_Styling_Criteria/en.md [hunk 17]
```diff
@@ -598,10 +509,10 @@ Try to limit writing text in images, as it is difficult for translators (and oth
 
 ##### Translating annotated images
 
-When translating annotated images, the localised image version must be placed in the same directory as the original version (i.e. the English version). The filename of a localised image version must start with the original version's name, followed by a hyphen, followed by the locale name (in caps). For example:
+When translating annotated images, the localised image version must be placed in the same directory as the original version (i.e. the English version). The filename of a localised image version must start with the original version's name, followed by a hyphen, followed by the locale name (in caps). For example:	
 
-- `hardrock_mod_vs_easy_mod.jpg` (i.e. in English)
-- `hardrock_mod_vs_easy_mod-DE.jpg` (i.e. in German)
+- `hardrock_mod_vs_easy_mod.jpg` (i.e. in English)	
+- `hardrock_mod_vs_easy_mod-DE.jpg` (i.e. in German)
```
Comment (Joehuu) [inline wiki/Article_Styling_Criteria/en.md]:
```suggestion
- `hardrock_mod_vs_easy_mod-DE.jpg` (i.e. in German)
```

**Comment by Joehuu** [comment]
Tournament tags are the exception of having duplicate keywords found in the article (content).

**Comment by cl8n** [comment]
if the keywords are important enough that they can identify the article, they're supposed to be in tags, regardless of if they already appear in content or not. tournaments wouldn't be the only exception

**Comment by sr229** [comment]
For greater clarity for this criteria, we can use the defining terms in [RFC 2119](https://tools.ietf.org/html/rfc2119).

**Comment by Joehuu** [comment]
Are the `nor` usages right throughout this article?

File: wiki/Article_Styling_Criteria/en.md [hunk 18]
```diff
@@ -298,60 +249,55 @@ Use italics only when naming a video game or the name of a work. osu!, the game,
...
-Optionally, add an extra line before and after the heading to help make the heading stand out while editing.
+Do not use headings to style or format text and limit to a heading level of 5. Use the [ATX (hash) style](https://github.github.com/gfm/#atx-headings). Each heading should be preceded and followed by an empty line.
 
 #### Titles
 
-*Notice: The title of an article must never be escaped, as titles are parsed as raw text, not Markdown.*
+*Notice: Article titles must not be escaped, as they are parsed as plain text, not Markdown.*
 
-All articles must have one level 1 heading, being the article's title. Article titles must use title case and never contain formatting, links, nor images. The title of an article is the name of the folder that the article is located in.
+The first heading in all articles must be a level 1 heading, being the article's title. All headings that follow the title heading must be [section headings](#sections). Article titles must use title case and must not contain formatting, links, nor images.
```
Comment (Joehuu) [inline wiki/Article_Styling_Criteria/en.md]:
```suggestion
The first heading in all articles must be a level 1 heading, being the article's title. All headings that follow the title heading must be [section headings](#sections). Article titles must use title case and must not contain formatting, links, or images.
```

File: wiki/Article_Styling_Criteria/en.md [hunk 19]
```diff
@@ -298,60 +249,55 @@ Use italics only when naming a video game or the name of a work. osu!, the game,
...
+The title of an article must match the folder name it is located in (spaces can replace underscores (`_`) where appropriate). The exception to this rule are tournament pages, which may use abbreviations, acronyms, or initialisms.
+
+---
+
+For English article titles, if the article's title needs to be renamed, the folder name it is in must also be renamed to match it.
 
 #### Sections
 
-All section headings must use levels 2 to 5, never level 6. Section headings must use sentence case and should never contain styles nor links. In addition to this, section headings must never skip a heading level (e.g. do not go from a level 2 heading to a level 4 heading). The first section heading must be a level 2 heading. Unlike titles, section headings can have small image icons.
+All section headings must use levels 2 to 5. Section headings must use sentence case and must not contain formatting nor links. In addition to this, section headings must not skip a heading level (e.g. do not go from a level 2 heading to a level 4 heading). The section heading after the [title heading](#titles) must be a level 2 heading. Unlike titles, section headings can have small image icons.
```
Comment (Joehuu) [inline wiki/Article_Styling_Criteria/en.md]:
```suggestion
All section headings must use levels 2 to 5. Section headings must use sentence case and must not contain formatting or links. In addition to this, section headings must not skip a heading level (e.g. do not go from a level 2 heading to a level 4 heading). The section heading after the [title heading](#titles) must be a level 2 heading. Unlike titles, section headings can have small image icons.
```

File: wiki/Article_Styling_Criteria/en.md [hunk 20]
```diff
@@ -727,7 +630,7 @@ For example, use:
 
 ### osu!
 
-The name of the game, osu!, is never capitalised nor italicised. The osu! official branding must not use any spaces. Examples include:
+The name of the game, osu!, must not be capitalised nor italicised. The osu! official branding must not use any spaces. Examples include:
```
Comment (Joehuu) [inline wiki/Article_Styling_Criteria/en.md]:
```suggestion
The name of the game, osu!, must not be capitalised or italicised. The osu! official branding must not use any spaces. Examples include:
```

File: wiki/Article_Styling_Criteria/en.md [hunk 21]
```diff
@@ -834,31 +737,33 @@ The `#spanish` chat channel are for those who speak Spanish.
...
 - `Google`
 - `Reddit`
 - `Skype`
-- `Twitter`
 - `Twitch`
+- `Twitter`
 - `YouTube`
 
-Never use the trademark nor registered trademark symbols with trademarks.
+Do not use the trademark nor registered trademark symbols with trademarks.
```
Comment (Joehuu) [inline wiki/Article_Styling_Criteria/en.md]:
```suggestion
Do not use the trademark or registered trademark symbols with trademarks.
```
